### ISSUE TITLE
[Bug]: Kafka Connectivity Error AWS MSK

### ISSUE THREAD

**Issue by roy-akash** [milvus-io/milvus#35682]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Environment

```markdown
- Milvus version: custom upon 2.3.19
- Deployment mode(standalone or cluster): helm
- MQ type(rocksmq, pulsar or kafka): kafka
- SDK version(e.g. pymilvus v2.0.0rc2): 2.3.5
- OS(Ubuntu or CentOS): RHEL
- CPU/Memory: 16, 64
- GPU: 0
- Others:
```

### Current Behavior

We have made a couple of changes in the base milvus tag 2.3.19 to support BYOK use case in our organisation. However these changes are unrelated to broker logic technically.

We are building are own milvus image with our own docker file that is based upon in house rhel9 OS and not ubuntu which official milvus uses.

With this new in house image we are facing kafka connectivity issue with AWS MSK, However, It is working fine on the custom image we build using official milvus docker image. Also we have made a few version changes for a few dependencies in the conanfile as well.

### Expected Behavior

Kafka connectivity should work 

### Steps To Reproduce

```markdown
Using our custom milvus image ingest and index the data
```

### Milvus Log

[milvus-log.tar.gz](https://github.com/user-attachments/files/16732591/milvus-log.tar.gz)
[rhel-dockerfile.txt](https://github.com/user-attachments/files/16732592/rhel-dockerfile.txt)
[docker-file.txt](https://github.com/user-attachments/files/16732599/docker-file.txt)

kafka config - 

kafka:
  brokerList: b-1.fabricmilvusmsk.uktx4o.c2.kafka.us-west-2.amazonaws.com:9096,b-3.fabricmilvusmsk.uktx4o.c2.kafka.us-west-2.amazonaws.com:9096,b-2.fabricmilvusmsk.uktx4o.c2.kafka.us-west-2.amazonaws.com:9096

     consumer:
    client.id: milvusbyokv8-c
  producer:
    client.id: milvusbyokv8-p

     saslUsername: fabric-milvus-msk-scram-username
  saslMechanisms: SCRAM-SHA-512

     securityProtocol: SASL_SSL

### Anything else?

I suspect there are certain certificates at milvus side which it might be implicitly getting from the ubuntu os to establish SASL_SSL with MSK brokers, this certificates could be missing in our inhouse RHEL9 OS. If you could. point to use which certs does kafka implementation in milvus uses it would be very helpful

**Comment by xiaofan-luan** [comment]
SSL handshake failed: error:0A000086:SSL routines::certificate verify failed

**Comment by yanliang567** [comment]
/assign @locustbaby 
please help to take a look
/unassign

**Comment by jaime0815** [comment]
Hi @roy-akash, Milvus will not get certificates from the OS in the default configuration. 
https://github.com/milvus-io/milvus/blob/master/pkg/mq/msgstream/mqwrapper/kafka/kafka_client.go#L104
```
	if config.KafkaUseSSL.GetAsBool() {
		kafkaConfig.SetKey("ssl.certificate.location", config.KafkaTLSCert.GetValue())
		kafkaConfig.SetKey("ssl.key.location", config.KafkaTLSKey.GetValue())
		kafkaConfig.SetKey("ssl.ca.location", config.KafkaTLSCACert.GetValue())
		if config.KafkaTLSKeyPassword.GetValue() != "" {
			kafkaConfig.SetKey("ssl.key.password", config.KafkaTLSKeyPassword.GetValue())
		}
	}
```
Did you update the librdkafka version in conanfile.py?  "There's a similar issue: https://stackoverflow.com/questions/76678569/confluent-kafka-python-certificate-verification

**Comment by roy-akash** [comment]
Hi @jaime0815 thanks for the suggestion, we were able to get this resolved after setting these properties explicitly in the milvus.yaml. 

There is a weird thing though, we did not set these values in the earlier version of our milvus deployment. We were definetly using SASL_SSL mode and things were working perfectly. This was the official image which i believe is build upon the ubuntu base image.

For our inhouse RHEL9 Based milvus image, we had to set these paths explicitly to resolve connectivity to MSK. 

Anyways our problem is solved and we can close this
